test/models/hymod/include/Et_Calc_Function_Test.cpp

```diff
@@ -0,0 +1,1179 @@
+  surf_rad_forcing.incoming_shortwave_radiation_W_per_sq_m= 
+          solar_results.solar_radiation_horizontal_cloudy_flux_W_per_sq_m;
+
+// UNIT TEST RESULTS
+// CALCULATED SOLAR FLUXES
+// at time:     20.56700000 UTC
+// at site latitude: 37.250000 deg. longitude:-97.555400 deg.  elevation:303.333000 m
+// Shortwave radiation clear-sky flux calculations:
+// -above canopy/snow perpendicular to Earth-Sun line is      =964.56166277 W/m2
+// -at the top of a horizontal canopy/snow surface is:        =661.40396086 W/m2
+// Shortwave radiation clear-sky flux calculations with 0.5000 cloud cover fraction:
+// -above canopy/snow perpendicular to Earth-Sun line is      =807.82039257 W/m2
+// -at the top of a horizontal canopy/snow surface is:        =553.92581722 W/m2
+// CALCULATED ANGLES DESCRIBING VECTOR POINTING TO THE SUN
+// solar elevation angle:     43.29101185 degrees
+// solar azimuth:            225.06371958 degrees
+// local hour angle:          31.01549773 degrees
+// Number of tests passed=7 of 7.
+// UNIT TEST PASSED.
+
+  }
+  
+// we must calculate the net radiation before calling the ET subroutine.
+if(et_options.use_aerodynamic_method==FALSE) 
+  {
+  // NOTE don't call this function use_aerodynamic_method option is TRUE
+  et_forcing.net_radiation_W_per_sq_m=calculate_net_radiation_W_per_sq_m(&et_options,&surf_rad_params, 
+                                                                         &surf_rad_forcing);
+  }
+
+if(et_options.use_energy_balance_method ==TRUE)
+  et_m_per_s=evapotranspiration_energy_balance_method(&et_options,&et_params,&et_forcing);
+if(et_options.use_aerodynamic_method ==TRUE)
+  et_m_per_s=evapotranspiration_aerodynamic_method(&et_options,&et_params,&et_forcing,&inter_vars);
+if(et_options.use_combination_method ==TRUE)
+  et_m_per_s=evapotranspiration_combination_method(&et_options,&et_params,&et_forcing,&inter_vars);
+if(et_options.use_priestley_taylor_method ==TRUE)
+  et_m_per_s=evapotranspiration_priestley_taylor_method(&et_options,&et_params,&et_forcing,&inter_vars);
+if(et_options.use_penman_monteith_method ==TRUE)
+  et_m_per_s=evapotranspiration_penman_monteith_method(&et_options,&et_params,&et_forcing,&inter_vars);
+
+if(et_options.use_energy_balance_method ==TRUE)   printf("energy balance method:\n");
+if(et_options.use_aerodynamic_method ==TRUE)      printf("aerodynamic method:\n");
+if(et_options.use_combination_method ==TRUE)      printf("combination method:\n");
+if(et_options.use_priestley_taylor_method ==TRUE) printf("Priestley-Taylor method:\n");
+if(et_options.use_penman_monteith_method ==TRUE)  printf("Penman Monteith method:\n");
+                                                 
+//printf("calculated instantaneous potential evapotranspiration (PET) =%8.6e m/s\n",et_m_per_s);
+//printf("calculated instantaneous potential evapotranspiration (PET) =%8.6lf mm/d\n",et_m_per_s*86400.0*1000.0);
+
+//EXPECT_DOUBLE_EQ (0.0, et_m_per_s-1.106268e-07);
```

Comment:
If the EXPECT_DOUBLE_EQ is giving problems with the precision, you can incorporate rounding like here:
https://github.com/NOAA-OWP/ngen/blob/master/test/forcing/Forcing_Test.cpp#L137